Урок 7. Непараметрические тесты
1-4 задачи решать с помощью функций, 5ю вручную

Выбрать тест и проверить, есть  ли различия между выборками:

1 )  Даны две  независимые выборки. Не соблюдается условие нормальности

x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции

2 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

3 ) Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

4) Даны 3 группы  учеников плавания. Сделайте вывод по различиям.

В 1 группе время на дистанцию 50 м составляют:

56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

5) Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

Выбрать тест и проверить, есть  ли различия между выборками:

1 )  Даны две  независимые выборки. Не соблюдается условие нормальности

x1  380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции


In [4]:
import numpy as np
import scipy.stats as st

In [5]:
alpha = 0.005

#Критерий Манна-Уитни
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

st.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Ответ: pvalue > alpha, следовательно статистически значимых различий нет.

2 ) Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150,  130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

In [6]:
alpha = 0.005

#Критерий Фридмана
dimension1 = np.array([150, 160, 165, 145, 155])
dimension2 = np.array([140, 155, 150,  130, 135])
dimension3 = np.array([130, 130, 120, 130, 125])

st.friedmanchisquare(dimension1, dimension2, dimension3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Ответ: pvalue > alpha, следовательно статистически значимых различий нет.

3 ) Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [15]:
#Критерий Уилкоксона
alpha = 0.005

dimension1 = np.array([150, 160, 165, 145, 155])
dimension2 = np.array([140, 155, 150,  130, 135])

d = dimension2 - dimension1

print(d)

st.wilcoxon(d)

[-10  -5 -15 -15 -20]


WilcoxonResult(statistic=0.0, pvalue=0.0625)

Ответ: pvalue > alpha, следовательно статистически значимых различий нет.

4) Даны 3 группы  учеников плавания. Сделайте вывод по различиям.

В 1 группе время на дистанцию 50 м составляют:

56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [16]:
#Критерий Крускала-Уоллиса
alpha = 0.005

group1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
group2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
group3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

st.kruskal(group1, group2, group3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

Ответ: pvalue > alpha, следовательно статистически значимых различий нет.

5) Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [18]:
alpha  = 0.005
N = 10
m = 2.5

#H_0 = mean = 2.5
#H_1 = mean != 2.5

array = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])

#Найдем выборочное среднее и стандартное отклонение
mean = array.mean()
std = array.std(ddof=1)
print(mean, std)

2.5279999999999996 0.1572542173961923


In [19]:
#Найдем значение t-статистики
t = (mean - 2.5) / std * np.sqrt(len(array))
print(t)

0.563061366180296


In [21]:
#Найдем критическое значение t-статистики с помощью таблицы распределения Стьюдента
tt = st.t.ppf(0.975, 9)
print(tt)

2.2621571627409915


Ответ: для критерия Стьюдента по таблице tt = 1.833 (для = 0.05 и 9 степенями свободы),
tt > t (1.833 > 0.56), данные не противоречат заявленной гипотезе о среднем арифметическом равном 2.5 см, гипотеза верна.